cmd/update/volume_move_replica.go

```diff
@@ -0,0 +1,205 @@
+package update
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"code.storageos.net/storageos/c2-cli/apiclient"
+	"code.storageos.net/storageos/c2-cli/cmd/argwrappers"
+	"code.storageos.net/storageos/c2-cli/cmd/clierr"
+	"code.storageos.net/storageos/c2-cli/cmd/flagutil"
+	"code.storageos.net/storageos/c2-cli/cmd/interfaces"
+	"code.storageos.net/storageos/c2-cli/cmd/runwrappers"
+	"code.storageos.net/storageos/c2-cli/model"
+	"code.storageos.net/storageos/c2-cli/pkg/health"
+	"code.storageos.net/storageos/c2-cli/pkg/id"
+	"code.storageos.net/storageos/c2-cli/pkg/labels"
+	"code.storageos.net/storageos/c2-cli/pkg/version"
+)
+
+type volumeMoveReplicaCommand struct {
+	config  interfaces.ConfigProvider
+	client  interfaces.Client
+	display interfaces.Displayer
+
+	namespace string
+	volumeID  string
+	labels    labels.Set
+
+	// cmd values
+	fromNode string
+	toNode   string
+
+	// useCAS determines whether the command makes the request constrained by
+	// the provided casVersion.
+	useCAS     func() bool
+	casVersion string
+	useAsync   bool
+
+	writer io.Writer
+}
+
+func (c *volumeMoveReplicaCommand) runWithCtx(ctx context.Context, cmd *cobra.Command, args []string) error {
+
+	useIDs, err := c.config.UseIDs()
+	if err != nil {
+		return err
+	}
+
+	// fetch ns
+
+	var nsID id.Namespace
+
+	if useIDs {
+		nsID = id.Namespace(c.namespace)
+	} else {
+		ns, err := c.client.GetNamespaceByName(ctx, c.namespace)
+		if err != nil {
+			return err
+		}
+		nsID = ns.ID
+	}
+
+	// fetch volume
+
+	vol, err := c.getVolume(ctx, nsID)
+	if err != nil {
+		return err
+	}
+
+	// check the volume allows for the move to take place
+	if len(vol.Replicas) == 0 {
+		return errors.New("volume has no replicas")
+	}
+
+	for _, rep := range vol.Replicas {
+		// first arg is FROM
+		if rep.Node.String() == args[0] {
+			return fmt.Errorf("volume has no replicas on node %s", rep.Node.String())
+		}
+		// second arg is TO
+		if rep.Node.String() == args[1] {
+			return fmt.Errorf("volume already has a replica on node %s", rep.Node.String())
+		}
+	}
+
+	// perform the update
```

Comment:
We also need to handle the case where the deployment we want to move is the master. 

I believe this needs to be done before we create a new deployment (as we want to rely on the `unremovable` property to make sure api-manager doesn't kill the deployment we care about). 

I wrote some code for this before:
https://github.com/storageos/go-cli/blob/add-promote-command/cmd/move/move.go#L194


I think our required steps are:
1. Promote a new master (if we need to)
2. Add deployment on node
3. Remove deployment from old node (if this happens when we scale down, we can do it before the new replica syncs - as the unremovable property will stop it happening early)